include/core/shape.h

```diff
@@ -207,9 +251,34 @@ class Shape : public AutoDerivativeFunction<dim>
    */
   Point<dim>
   align_and_center(const Point<dim> &evaluation_point) const;
+
+  /**
+   * @brief
+   * This function applies the inverse operation of align_and_center
+   *
+   * Returns the centered and aligned point used on the levelset evaluation in
+   * the global reference frame.
+   *
+   * @param evaluation_point The point that will be centered and aligned in the global reference frame.
+   * @return The aligned and centered point
+   */
+  Point<dim>
+  reverse_align_and_center(const Point<dim> &evaluation_point) const;
+
+  /**
+   * @brief
+   * This function returns a point in a string of text. This is used in the
+   * cache of the shape.
+   *
+   * @param evaluation_point is the point that is transformed to its text form.
+   */
+  std::string
+  point_to_string(const Point<dim> &evaluation_point) const;
   // Effective radius used for crown refinement
   double effective_radius;
 
+  // String that contains additional information on the shape
+  std::string additional_info_on_shape;
```

Comment:
Does this only indicates the type of the cad file (iges, step or stl)? If so, maybe a more precise comment and/or variable name could be nice :)